aiida_workgraph/socket.py

```diff
@@ -10,6 +10,41 @@ class TaskSocket(NodeSocket):
     # to override the default NodeProperty from node_graph
     node_property = TaskProperty
 
+    @property
+    def node_value(self):
+        if not hasattr(self, '_node_value'):
+            self._node_value = self.get_node_value()
+        return self._node_value
+
+    def get_node_value(self):
+        "Directly return or set the _node_value attribute."
+
+        # _node_value was set before already
+        if hasattr(self, '_node_value'):
+            pass
+
+        # If data associated with Socket is AiiDA ORM, return again its value
+        # Check for the nested case before, otherwise Socket value is matched
+        # first and the ORM instance returned
+        elif hasattr(self.value, "value"):
+            # TODO: One could also check for isinstance of AiiDA ORM, however, that adds another import, and not sure if
+            # TODO: it's really necessary here
+
+            self._node_value = self.value.value
+            # TODO: Possibly check here for AttributeDict, for which we return the `get_dict` result directly
+            # TODO: However, with a specific `SocketAiiDADict` class, we could just overwrite the method there
+
+        # If not, e.g. when Python base types are used, directly return the variable's value
+        elif hasattr(self, "value"):
+            self._node_value = self.value
+
+        # TODO: This shouldn't even be a case? Shouldn't a `Socket` _always_ have an associated value, even if it's
+        # TODO: `None` on instantiation
+        else:
+            msg = f"Socket <{self}> does not have an asociated `value` attribute."
+            raise AttributeError(msg)
+
+        return self._node_value
```

Comment:
```suggestion
    """Retrieve the raw Python value from an AiiDA ORM data node.
    This method should be implemented in subclasses to handle specific socket types.
    """
    raise NotImplementedError("The method 'get_node_value' must be implemented in the subclass.")
```

And in the `SocketAny`, you can implement something similar to:

```python
    if isinstance(self.value, orm.Data):
        if hasattr(self.value, "value"):
            return self.value.value
        else:
            raise ValueError("Data node does not have a value attribute. We do not know how to extract the raw Python value.")
    else:
        return self.value
```